static/js/src/ua-payment-modal.js

```diff
@@ -316,35 +323,49 @@ function applyTotals() {
     country: country,
   };
 
-  postCustomerInfoForPurchasePreview(
-    currentTransaction.accountId,
-    addressObject,
-    taxObject
-  )
-    .then(() => {
-      if (currentTransaction.type === "purchase") {
-        postPurchasePreviewData(
-          currentTransaction.accountId,
-          currentTransaction.products,
-          currentTransaction.previousPurchaseId
-        ).then((data) => {
-          purchasePreview = data;
-          modal.classList.remove("is-processing");
-          setOrderTotals(country, vatApplicable, purchasePreview, modal);
-        });
-      } else if (currentTransaction.type === "renewal") {
-        postRenewalPreviewData(currentTransaction.transactionId).then(
-          (data) => {
+  if (currentTransaction.accountId) {
+    postCustomerInfoForPurchasePreview(
+      currentTransaction.accountId,
+      addressObject,
+      taxObject
+    )
+      .then(() => {
+        if (currentTransaction.type === "purchase") {
+          postPurchasePreviewData(
+            currentTransaction.accountId,
+            currentTransaction.products,
+            currentTransaction.previousPurchaseId
+          ).then((data) => {
             purchasePreview = data;
             modal.classList.remove("is-processing");
             setOrderTotals(country, vatApplicable, purchasePreview, modal);
-          }
-        );
-      }
-    })
-    .catch((error) => {
-      console.error(error);
-    });
+          });
+        } else if (currentTransaction.type === "renewal") {
+          postRenewalPreviewData(currentTransaction.transactionId).then(
```

Comment:
The renewal preview is only valid if you are accepting renewals as purchases, which I think is not being done yet, right? In that case, I'd recommend against including it for now.

Reply:
Ah OK - I'll update